src/sources/soundsourceproxy.cpp

```diff
@@ -267,14 +270,46 @@ bool SoundSourceProxy::isFileNameSupported(const QString& fileName) {
 }
 
 // static
-bool SoundSourceProxy::isFileExtensionSupported(const QString& fileExtension) {
-    return !s_soundSourceProviders.getRegistrationsForFileExtension(fileExtension).isEmpty();
+bool SoundSourceProxy::isFileTypeSupported(const QString& fileType) {
+    return !s_soundSourceProviders.getRegistrationsForFileType(fileType).isEmpty();
+}
+
+// static
+bool SoundSourceProxy::isFileSuffixSupported(const QString& fileSuffix) {
+    return getSupportedFileSuffixes().contains(fileSuffix);
+}
+
+//static
+mixxx::SoundSourceProviderPointer SoundSourceProxy::getPrimaryProviderForFileType(
+        const QString& fileType) {
+    return s_soundSourceProviders.getPrimaryProviderForFileType(fileType);
 }
 
 //static
-mixxx::SoundSourceProviderPointer SoundSourceProxy::getPrimaryProviderForFileExtension(
-        const QString& fileExtension) {
-    return s_soundSourceProviders.getPrimaryProviderForFileExtension(fileExtension);
+QStringList SoundSourceProxy::getFileSuffixesForFileType(
+        const QString& fileType) {
+    // Each file type is a valid file suffix
+    const QString dummyFileName = QStringLiteral("prefix.") + fileType;
+    const auto mimeTypes = QMimeDatabase().mimeTypesForFileName(dummyFileName);
+    QStringList fileSuffixes;
+    fileSuffixes.reserve(mimeTypes.size() * 2);
+    for (const QMimeType& mimeType : mimeTypes) {
+        fileSuffixes.append(mimeType.suffixes());
+    }
+    fileSuffixes.removeDuplicates();
+    return fileSuffixes;
```

Comment:
do we want to handle files without a suffix? My guess is that this is a common case on windows where the file extension is hidden.

Reply:
We probably never handled files without a suffix correctly and never tested it. I don't want to extend the scope of this PR.

Reply:
OK